routing/pathfind.go

```diff
@@ -585,10 +590,79 @@ func findPaths(tx *bolt.Tx, graph *channeldb.ChannelGraph,
 		candidatePaths pathHeap
 	)
 
+	var err error
+	if tx == nil {
+		tx, err = graph.Database().Begin(false)
+		if err != nil {
+			return nil, err
+		}
+		defer tx.Rollback()
+	}
+
+	// Iterate over all nodes in the graph database, and add them to an
+	// array which can back an in-memory graphSource. We do this to avoid
+	// having to read from the database during each call, speeding up the
+	// execution significantly.
+	var nodes []*channeldb.LightningNode
+	if err := graph.ForEachNode(nil,
```

Comment:
This is hard to visually parse. Can you separate the err assignment from the if?

```go
err = graph.ForEachNode(nil, fun(...) {
    ...
})
if err != nil {
    return nil, err
}
```